gh-ci-pipeline/src/main/java/Main.java

```diff
@@ -0,0 +1,190 @@
+
+import javax.crypto.Cipher;
+import javax.crypto.NoSuchPaddingException;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.nio.ByteBuffer;
+import java.security.NoSuchAlgorithmException;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import com.amazonaws.services.kms.AWSKMS;
+import com.amazonaws.services.kms.AWSKMSClientBuilder;
+import com.amazonaws.services.kms.model.DecryptRequest;
+import com.amazonaws.services.kms.model.EncryptRequest;
+import com.amazonaws.services.kms.model.ReEncryptRequest;
+
+public class Main {
+
+    // From slides -------------------------------------
+    static String myKeyId = "AKIAX742FUDUQXXXXXXX";
+    static String mySecretKey = "MY_SECRET_KEY";
+    static final String CIPHER = "DES";
+    static final String CIPHER_GOOD ="RSA/ECB/OAEPPadding";
+
+    static final String VALID_PATH1 = "./test/file1.txt";
+    static final String VALID_PATH2 = "./test/file2.txt";
+    static final String DEFAULT_VALID_PATH = "./test/file3.txt";
+
+    public static void main(String[] args) {
+        AWSCredentials creds =
+            getCreds(myKeyId, mySecretKey);
+        System.out.println(creds.getAWSSecretKey());
+
+        AWSCredentials creds2 = getCreds();
+        System.out.println(creds2.getAWSSecretKey());
+
+        run1();
+        run2();
+
+        reEncrypt1();
+        reEncrypt2();
+    }
+
+    // Bad
+    static AWSCredentials getCreds(String id, String key) {
+        return new BasicAWSCredentials(id, key);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

If possible, use temporary security credentials (IAM roles) instead of long-term access keys.
Long-term access keys, such as those associated with IAM users and AWS account root users, remain valid until you manually revoke them. However, temporary security credentials obtained through IAM roles and other features of the AWS Security Token Service expire after a short period of time. Use temporary security credentials to help reduce your risk in case credentials are accidentally exposed.
Learn more about [best practices for managing AWS access keys](https://docs.aws.amazon.com/general/latest/gr/aws-access-keys-best-practices.html).